pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -424,7 +424,8 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
     }
 
     /**
-     * @dev Process a GivenIn swap involving BPT.
+     * @dev Process a GivenIn swap involving BPT. At this point, amount has been upscaled, and the BPT token
+     * has been removed from balances. `indexIn` and `indexOut` include BPT.
```

Comment:
Could we drop bpt from these? It is slightly wasteful as we only ever use only one of the two indices, but it's a cheap operation and will make this function more consistent (i.e. not have to be aware of bpt shenanigans).

Reply:
The trouble is we don't know which one we need until pretty late in the process. It makes it asymmetric with swaps (those call the base class, so we can't affect those), but I can do it on the swapWithBpt top level (which is symmetric with onRegularSwap)

Reply:
Ok - next problem is this conflicts with the range checking (that skipBpt can't be given a BPT value). The code works now because we only call _skipBptIndex when it's not BPT. But if we call it in the top-level function, we know one of them is going to be BPT so it will fail.

Explaining and pre-calculating all that is probably worse than leaving it as is. (Or fixing it and not having the input range check on _skipBptIndex.)

Reply:
Hm I think you're right. Let's keep as is then.